WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostDetailFragment.java

```diff
@@ -520,40 +539,84 @@ public void onEventMainThread(ReaderEvents.RelatedPostsUpdated event) {
         relatedPostsView.setOnRelatedPostClickListener(new ReaderRelatedPostsView.OnRelatedPostClickListener() {
             @Override
             public void onRelatedPostClick(View v, long siteId, long postId) {
-                showRelatedPostDetail(siteId, postId);
+                showRelatedPostDetail(siteId, postId, isGlobal);
             }
         });
 
+        // fade in this related posts view
         if (relatedPostsView.getVisibility() != View.VISIBLE) {
             AniUtils.fadeIn(relatedPostsView, AniUtils.Duration.MEDIUM);
         }
+
+        trackRelatedPostsIfShowing();
+    }
+
+    /*
+     * track that related posts have loaded and are scrolled into view if we haven't
+     * already tracked it
+     */
+    private void trackRelatedPostsIfShowing() {
+        if (!mHasTrackedGlobalRelatedPosts && isVisibleAndScrolledIntoView(mGlobalRelatedPostsView)) {
+            mHasTrackedGlobalRelatedPosts = true;
+            AppLog.d(T.READER, "reader post detail > global related posts rendered");
+            mGlobalRelatedPostsView.trackRailcarRender();
+        }
+
+        if (!mHasTrackedLocalRelatedPosts && isVisibleAndScrolledIntoView(mLocalRelatedPostsView)) {
+            mHasTrackedLocalRelatedPosts = true;
+            AppLog.d(T.READER, "reader post detail > local related posts rendered");
+            mLocalRelatedPostsView.trackRailcarRender();
+        }
+    }
+
+    /*
+     * returns True if the passed view is visible and has been scrolled into view - assumes
+     * that the view is a child of mScrollView
+     */
+    private boolean isVisibleAndScrolledIntoView(View view) {
+        if (view != null && view.getVisibility() == View.VISIBLE) {
+            Rect scrollBounds = new Rect();
+            mScrollView.getHitRect(scrollBounds);
+            return view.getLocalVisibleRect(scrollBounds);
+        } else {
+            return false;
+        }
     }
 
     /*
      * user clicked a single related post - if we're already viewing a related post, add it to the
      * history stack so the user can back-button through the history - otherwise start a new detail
      * activity for this related post
      */
-    private void showRelatedPostDetail(long blogId, long postId) {
-        AnalyticsUtils.trackWithReaderPostDetails(
-                AnalyticsTracker.Stat.READER_RELATED_POST_CLICKED, blogId, postId);
+    private void showRelatedPostDetail(long blogId, long postId, boolean isGlobal) {
+        AnalyticsTracker.Stat stat = isGlobal
+                ? AnalyticsTracker.Stat.READER_GLOBAL_RELATED_POST_CLICKED
+                : AnalyticsTracker.Stat.READER_LOCAL_RELATED_POST_CLICKED;
+        AnalyticsUtils.trackWithReaderPostDetails(stat, blogId, postId);
 
         if (mIsRelatedPost) {
             mPostHistory.push(new ReaderBlogIdPostId(mPost.blogId, mPost.postId));
-            replacePost(blogId, postId, true);
+            replacePost(blogId, postId, false);
```

Comment:
I am not sure if I understand the reason of this change and the following. I looked into the commit but it's titled `Resolved conflicts`, so do you mind explaining the changes here just to make sure I don't miss anything. Thanks!